restapi/src/main/java/io/stargate/web/docsapi/resources/CollectionsResource.java

```diff
@@ -83,20 +87,25 @@ public Response getCollections(
       @Context HttpServletRequest request) {
     return RequestHandler.handle(
         () -> {
-          Map<String, String> allHeaders = getAllHeaders(request);
-          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, allHeaders);
-          Collection<Table> tables = authenticatedDB.getTables(namespace);
+          DocumentDB docDB = db.getDocDataStoreForToken(token, getAllHeaders(request));
+          Keyspace ks = docDB.schema().keyspace(namespace);
+          if (ks == null) {
+            throw new NotFoundException(String.format("keyspace '%s' not found", namespace));
+          }
+          Collection<Table> tables = ks.tables();
 
           db.getAuthorizationService()
               .authorizeSchemaRead(
-                  authenticatedDB.getAuthenticationSubject(),
+                  docDB.getAuthenticationSubject(),
                   Collections.singletonList(namespace),
                   tables.stream().map(SchemaEntity::name).collect(Collectors.toList()),
                   SourceAPI.REST,
                   ResourceKind.TABLE);
 
           List<DocCollection> result =
               tables.stream()
+                  .filter(
```

Comment:
Should we filter first and then authenticate? Doesn't seem like it would make sense to return a 403 for a keyspace that isn't being returned anyway.

Reply:
Right, ofc - this should be done in opposite order...

Reply:
fixed